tests/Unit/Helpers/Evolution/DiscontinuousGalerkin/BoundaryCorrections.hpp

```diff
@@ -0,0 +1,326 @@
+      "listed in the system as being primitive variables");
+
+  MAKE_GENERATOR(gen);
+  std::uniform_real_distribution<> dist(0.0, 1.0);
+  std::uniform_real_distribution<> pos_neg_dist(-1.0, 1.0);
+  DataVector used_for_size{face_mesh.number_of_grid_points()};
+
+  boost::optional<tnsr::I<DataVector, FaceDim + 1, Frame::Inertial>>
+      mesh_velocity{};
+  if (use_moving_mesh) {
+    mesh_velocity = make_with_random_values<
+        tnsr::I<DataVector, FaceDim + 1, Frame::Inertial>>(
+        make_not_null(&gen), make_not_null(&dist), used_for_size);
+  }
+
+  Variables<dg_package_field_tags> interior_package_data{used_for_size.size()};
+  const auto interior_fields_on_face = make_with_random_values<
+      Variables<tmpl::append<variables_tags, flux_tags, package_temporary_tags,
+                             package_primitive_tags>>>(
+      make_not_null(&gen), make_not_null(&dist), used_for_size);
+
+  Variables<dg_package_field_tags> exterior_package_data{used_for_size.size()};
+  const auto exterior_fields_on_face = make_with_random_values<
+      Variables<tmpl::append<variables_tags, flux_tags, package_temporary_tags,
+                             package_primitive_tags>>>(
+      make_not_null(&gen), make_not_null(&dist), used_for_size);
+
+  // Compute the interior and exterior normal vectors so they are pointing in
+  // opposite directions.
+  auto interior_unit_normal_covector = make_with_random_values<
+      tnsr::i<DataVector, FaceDim + 1, Frame::Inertial>>(
+      make_not_null(&gen), make_not_null(&pos_neg_dist), used_for_size);
+  const Scalar<DataVector> interior_normal_magnitude =
+      magnitude(interior_unit_normal_covector);
+  for (auto& t : interior_unit_normal_covector) {
+    t /= get(interior_normal_magnitude);
+  }
+  auto exterior_unit_normal_covector = interior_unit_normal_covector;
+  for (auto& t : exterior_unit_normal_covector) {
+    t *= -1.0;
+  }
+
+  call_dg_package_data(make_not_null(&interior_package_data), correction,
+                       interior_fields_on_face, volume_data,
+                       interior_unit_normal_covector, mesh_velocity);
+  call_dg_package_data(make_not_null(&exterior_package_data), correction,
+                       exterior_fields_on_face, volume_data,
+                       exterior_unit_normal_covector, mesh_velocity);
+
+  Variables<dt_variables_tags> boundary_corrections{
+      face_mesh.number_of_grid_points()};
+  call_dg_boundary_terms(make_not_null(&boundary_corrections), correction,
+                         interior_package_data, exterior_package_data,
+                         dg_formulation);
+
+  if (dg_formulation == ::dg::Formulation::StrongInertial) {
+    // The strong form should be (WeakForm - (n_i F^i)_{interior}).
+    // Since we test conservation for the weak form we just need to test that
+    // the strong form satisfies the above definition.
+
+    call_dg_package_data(make_not_null(&interior_package_data), correction,
```

Comment:
It appears that the `...package_data` variables should already have the values you want from the previous calls.